livedata/src/main/java/io/getstream/chat/android/livedata/ChatDomain.kt

```diff
@@ -108,9 +108,11 @@ interface ChatDomain {
 
         fun backgroundSyncEnabled(apiKey: String, userToken: String): Builder {
             // TODO: Consider exposing apiKey and userToken by ChatClient to make this function more friendly
-            this.backgroundSyncConfig = BackgroundSyncConfig(apiKey, user.id, userToken)
             if (apiKey.isEmpty() || user.id.isEmpty() || userToken.isEmpty()) {
-                throw IllegalArgumentException("ChatDomain.Builder::backgroundSyncEnabled. apiKey and user.id, and userToken must not be empty.")
+                this.backgroundSyncConfig = BackgroundSyncConfig.UNAVAILABLE
+                throw IllegalArgumentException("ChatDomain.Builder::backgroundSyncEnabled. apiKey, user.id, and userToken must not be empty.")
```

Comment:
Do we really expect clients to catch the exception and then proceed to use the same `Builder`? If they do something invalid with it, I think we should mark that instance as broken and never let it `build`. That's something they should fix in their client code, they shouldn't get away with it 😄

Reply:
In case Builder instance is "broken" would you throw an exception inside `build()` instead? Or do you have some other idea of preventing instantiation of ChatDomain 🤔

Reply:
Yup, that's what I was thinking. Somehow invalidate it to prevent anyone from using it, if they've ever made an invalid call to it.